chat-android/src/main/java/com/ably/chat/RoomStatus.kt

```diff
@@ -186,32 +200,37 @@ class DefaultStatus(private val logger: LogHandler? = null) : InternalRoomStatus
     override val error: ErrorInfo?
         get() = _error
 
-    private val internalEmitter = RoomStatusEvenEmitter()
+    private val internalEmitter = RoomStatusInternalAsyncEventEmitter(roomScope)
```

Comment:
Currently, `internalEmitter` is implemented as per chat-js, but in the future, can be moved inside `RoomLifecycleManager`, since it is only used by `RoomLifecycleManager`